client/post-editor/editor-sharing/publicize-services.jsx

```diff
@@ -1,59 +1,72 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { filter, uniqBy } from 'lodash';
 
 /**
  * Internal dependencies
  */
-var serviceConnections = require( 'my-sites/sharing/connections/service-connections' ),
-	EditorSharingPublicizeConnection = require( './publicize-connection' );
+import EditorSharingPublicizeConnection from './publicize-connection';
+import { getCurrentUserId } from 'state/current-user/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteUserConnections } from 'state/sharing/publicize/selectors';
 
-module.exports = React.createClass( {
-	displayName: 'EditorSharingPublicizeServices',
+class EditorSharingPublicizeServices extends Component {
+	static propTypes = {
+		post: PropTypes.object,
+		connections: PropTypes.array.isRequired,
+		newConnectionPopup: PropTypes.func.isRequired
+	};
 
-	propTypes: {
-		post: React.PropTypes.object,
-		siteId: React.PropTypes.number.isRequired,
-		connections: React.PropTypes.array.isRequired,
-		newConnectionPopup: React.PropTypes.func.isRequired
-	},
+	static defaultProps = {
+		post: Object.freeze( {} ),
```

Comment:
My rule-of-thumb has been: If a prop is not used by the component itself (i.e. just passed to a descendant) then put the burden of defining a default on the component where it's being used. Otherwise it becomes quite verbose and, really, there's no need for a default if the component itself doesn't care what the value is.

Also, I don't know that we need `Object.freeze`. It helps avoid accidental mutations to props, but if we're doing that, then we have bigger problems anyways.